src/background/permissions/permissions-ensurer.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+
+export class PermissionsEnsurer {
+    public static noPermissionsMessageEnding = 'Extension manifest must request permission to access this host.';
+    public static noUserGestureMessage = 'This function must be called during a user gesture';
+
+    constructor(private readonly browserAdapter: BrowserAdapter) {}
+
+    public async ensureInjectPermissions(tabId: number): Promise<boolean> {
```

Comment:
It feels a little off for the caller to have to understand 2 different error paths (resolve(false) vs reject(error)). Is there a use case we're anticipating where we need to respond differently to those 2 cases?

Reply:
(if not, maybe `Promise<void>` would be more appropriate)